File: timefusion/desktop-app/src/main/java/model/Team.java
package main.java.model;

import java.util.List;

public class Team {

  private Long id;
  private String name;
  private String createdAt;
  private User leader; // Un utilisateur spécifique comme manager
  private List<User> coleaders; // Liste des membres de l'équipe
  private List<User> elders; // Liste des membres de l'équipe
  private List<User> members; // Liste des membres de l'équipe

  // Constructeurs, getters, setters

  public Team(
    Long id,
    String name,
    String createdAt,
    User leader,
    List<User> coleaders,
    List<User> elders,
    List<User> members
  ) {
    this.id = id;
    this.name = name;
    this.createdAt = createdAt;
    this.leader = leader;
    this.coleaders = coleaders;
    this.elders = elders;
    this.members = members;
  }

  public Long getId() {
    return this.id;
  }

  public String getName() {
    return this.name;
  }

  public String getCreatedAt() {
    return this.createdAt;
  }

  public User getLeader() {
    return this.leader;
  }

  public List<User> getColeaders() {
    return this.coleaders;
  }

  public List<User> getElders() {
    return this.elders;
  }

  public List<User> getMembers() {
    return this.members;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public void setName(String name) {
    this.name = name;
  }

  public void setCreatedAt(String createdAt) {
    this.createdAt = createdAt;
  }

  public void setLender(User leader) {
    this.leader = leader;
  }

  public void setColeaders(List<User> coleaders) {
    this.coleaders = coleaders;
  }

  public void setElders(List<User> elders) {
    this.elders = elders;
  }

  public void setMembers(List<User> members) {
    this.members = members;
  }
}

File: timefusion-web/src/main/webapp/PHP/src/Team/Team.php
<?php

namespace TimeFusion\Team;

class Team
{
    private $members = [];
    private $name;

    private $color;
    
    private $id;

    public function __construct(string $name = null, array $members = [], string $color = "#000352")
    {
        $this->name = $name;
        $this->members = $members;
        $this->color = $color;
    }

    public static function createFromDbResult(array $result): Team
    {
        $team = new self($result['team_name'], $result['team_members']);
        $team->setId($result['id']);

        return $team;
    }

    // Ajouter un membre à l'équipe avec un rôle spécifique
    public function addMember($user, $role)
    {
        $this->members[] = ['user' => $user, 'role' => $role];
    }

    // Obtenir la liste des membres de l'équipe avec leurs rôles
    public function getMembers()
    {
        return $this->members;
    }

    public function getId()
    {
        return $this->id;
    }

    public function getColor(){
        return $this->color;
    }

    public function getName()
    {
        return $this->name;
    }

    public function setId($id)
    {
        $this->id = $id;
    }

    public function setName(string $name){
        $this->name = $name;
    }

    public function setColor(string $color){
        $this->color = $color;
    }

    public function setMembers($members)
    {
        $this->members = $members;
    }

}
